ios/FluentUI/Navigation/NavigationBar.swift

```diff
@@ -107,43 +107,11 @@ open class NavigationBar: UINavigationBar, TokenizedControlInternal, TwoLineTitl
         case primary
         case system
         case custom
-
-        func tintColor(fluentTheme: FluentTheme) -> UIColor {
-            switch self {
-            case .primary, .default, .custom:
-                return UIColor(light: fluentTheme.color(.foregroundOnColor).light,
-                               dark: fluentTheme.color(.foreground2).dark)
-            case .system:
-                return fluentTheme.color(.foreground2)
-            }
-        }
-
-        func titleColor(fluentTheme: FluentTheme) -> UIColor {
-            switch self {
-            case .primary, .default, .custom:
-                return UIColor(light: fluentTheme.color(.foregroundOnColor).light,
-                               dark: fluentTheme.color(.foreground1).dark)
-            case .system:
-                return fluentTheme.color(.foreground1)
-            }
-        }
-
-        public func backgroundColor(fluentTheme: FluentTheme, customColor: UIColor? = nil) -> UIColor {
-            let defaultColor = UIColor(light: fluentTheme.color(.brandBackground1).light,
-                                       dark: fluentTheme.color(.background3).dark)
-            switch self {
-            case .primary, .default:
-                return defaultColor
-            case .system:
-                return fluentTheme.color(.background3)
-            case .custom:
-                return customColor ?? defaultColor
-            }
-        }
     }
 
+    @available(*, deprecated, renamed: "backgroundColor(forStyle:theme:)", message: "NavigationBar has been tokenized. Use the replacement method, passing in .system for the style.")
     @objc public static func navigationBarBackgroundColor(fluentTheme: FluentTheme) -> UIColor {
```

Comment:
Could you make this an an optional `FluentTheme` and use the shared theme as fallback (#1673 for reference)

Reply:
Ooh, good idea! Done with a7a741b.